src/coreclr/tools/aot/ILCompiler.Diagnostics/PdbWriter.cs

```diff
@@ -136,15 +136,18 @@ public void WritePDBData(string dllPath, IEnumerable<MethodInfo> methods)
             bool failed = true;
             try
             {
-                try
+                using (_ngenWriter)
```

Comment:
`_ngenWriter` is assigned in `WritePDBDataHelper`, so I don't know if this will do the right thing. Maybe we need to just manually call `Dispose` in the finally block after `CloseMod` is called?